src/Sample.Common/PromotheusExporterHostedService.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry.Exporter.Prometheus;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Metrics.Configuration;
+using OpenTelemetry.Metrics.Export;
+
+namespace Sample.Common
+{
+    public class PromotheusExporterHostedService : IHostedService
+    {
+        private readonly PrometheusExporter exporter;
+        private readonly IEnumerable<IAppMetrics> initializers;
+        private Timer timer;
+        private MeterFactory meterFactory;
+
+        public PromotheusExporterHostedService(PrometheusExporter exporter, IEnumerable<IAppMetrics> initializers)
+        {
+            this.exporter = exporter ?? throw new System.ArgumentNullException(nameof(exporter));
+            this.initializers = initializers;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            var interval = TimeSpan.FromSeconds(5);
+            var simpleProcessor = new UngroupedBatcher(exporter, interval);
+            this.meterFactory = MeterFactory.Create(simpleProcessor);
+
+            foreach (var initializer in initializers)
+            {
+                initializer.Initialize(meterFactory);
+            }
+
+            this.timer = new Timer(CollectMetrics, meterFactory, Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+
+            exporter.Start();
+
+            this.timer.Change(interval, interval);
```

Comment:
@SergeyKanzhelev 
I had to add this timer here in order to make the Prometheus exporter work. What is the correct way to make it work?